### PR TITLE
Created "Rect" widget, included in example.  😎🔫

### PR DESCRIPTION
**Pull request by Zekiah-A**
Widget for a box shape. Put a demo in an example. Also fixed some **wrong** spelling of certain words.

### CODE REVIEW

**Comment by Zekiah-A** [comment]
I wrote half of this in windows notepad, hopefully it compiles 😅

**Comment by Zekiah-A** [comment]
SCREW NOTEPAD
OK DON'T ACCEPT YET OK IT'S BROKEN OK

File: GUILIB/Styles/Buttons/ButtonStyle.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@ namespace GUILIB.Styles.Buttons
 {
     public class ButtonStyle
     {
-        #region Colors
+        #region Colours
```
Comment (StarlkYT) [inline GUILIB/Styles/Buttons/ButtonStyle.cs:7]: 😎

Comment (Zekiah-A) [reply GUILIB/Styles/Buttons/ButtonStyle.cs:7]: Bruh

Comment (Zekiah-A) [reply GUILIB/Styles/Buttons/ButtonStyle.cs:7]: Let me fix this, STUPID windows notepad broke my code lol

**Review by StarlkYT — CHANGES_REQUESTED** [review]
Looks good. But in `RectWidgte.cs` you used "_posX", "_width" etc... Can you remove all of them and replace it with a `Rectangle` then use `DrawRectangleRec`, this would be easier.

File: GUILIB/Styles/Buttons/ButtonStyle.cs (hunk 1 above)
Comment (Zekiah-A) [reply GUILIB/Styles/Buttons/ButtonStyle.cs:7]: > 😎

image

**Comment by Zekiah-A** [comment]
> Looks good. But in `RectWidgte.cs` you used "_posX", "_width" etc... Can you remove all of them and replace it with a `Rectangle` then use `DrawRectangleRec`, this would be easier.

Yeah i was doing this right now

**Comment by StarlkYT** [comment]
> > Looks good. But in `RectWidgte.cs` you used "_posX", "_width" etc... Can you remove all of them and replace it with a `Rectangle` then use `DrawRectangleRec`, this would be easier.
> 
> Yeah i was doing this right now

Let me know when you finish

**Comment by Zekiah-A** [comment]
> > > Looks good. But in `RectWidgte.cs` you used "_posX", "_width" etc... Can you remove all of them and replace it with a `Rectangle` then use `DrawRectangleRec`, this would be easier.
> > 
> > 
> > Yeah i was doing this right now
> 
> Let me know when you finish

ok i will ok

**Review by StarlkYT — CHANGES_REQUESTED** [review]
Perfection. Only very little things to change, I will do them myself.

**Comment by Zekiah-A** [comment]
> > > > Looks good. But in `RectWidgte.cs` you used "_posX", "_width" etc... Can you remove all of them and replace it with a `Rectangle` then use `DrawRectangleRec`, this would be easier.
> > > 
> > > 
> > > Yeah i was doing this right now
> > 
> > 
> > Let me know when you finish
> 
> ok i will ok

BRUH

**Comment by Zekiah-A** [comment]
> > > > > Looks good. But in `RectWidgte.cs` you used "_posX", "_width" etc... Can you remove all of them and replace it with a `Rectangle` then use `DrawRectangleRec`, this would be easier.
> > > > 
> > > > 
> > > > Yeah i was doing this right now
> > > 
> > > 
> > > Let me know when you finish
> > 
> > 
> > ok i will ok
> 
> BRUH

I just finished changing it 😢🔫    😢🔫    😢🔫    😢🔫    😢🔫    😢🔫    😢🔫    😢🔫

**Comment by Zekiah-A** [comment]
lol cool

**Pull request merged**